[Config] keep unused nested dict configs in storage

techfreaque opened the pull request:
- when nested object configs are used, previously a change in the object settings structure did overwrite existing unused configs. With ndict, we'll keep old unused nested configs, in case the user switches back to the previous object settings tree, to still have all the settings filled out as before.
- add keep_existing config option to define which way to use

GuillaumeDSM commented:
Thanks for the improvement @techfreaque !
Can we not use a lib for that though please ? I see it's a 1 star lib on github https://github.com/eiphy/naapc. Each lib we add is adding risks into the software and we prefer limiting it to well known libs only.
I think we should be able to acheive the same result with a few lines of code

techfreaque commented:
> Thanks for the improvement @techfreaque ! Can we not use a lib for that though please ? I see it's a 1 star lib on github https://github.com/eiphy/naapc. Each lib we add is adding risks into the software and we prefer limiting it to well known libs only. I think we should be able to acheive the same result with a few lines of code

Oh yeah you're right, I changed it and added tests

GuillaumeDSM commented on tests/configuration/test_tentacle_configuration.py:
```diff
@@ -67,6 +67,159 @@ async def test_update_config():
+    # init nested config
+    config_update = {
+        "price_difference_threshold_percent": 2,
+        "plop": 42,
```
😂

GuillaumeDSM commented on tests/configuration/test_tentacle_configuration.py:
```diff
@@ -67,6 +67,159 @@ async def test_update_config():
+            }
+        }
+    }
+    _cleanup()
```
great tests, thanks @techfreaque !
